pkg/k8s/k8s.go

```diff
@@ -513,3 +509,43 @@ func setupCacheVolume(cache v1alpha1.NginxCacheSpec, dep *appv1.Deployment) {
 		MountPath: cache.Path,
 	})
 }
+
+func setupLifecycle(lifecycle *corev1.Lifecycle, dep *appv1.Deployment) {
+	if lifecycle == nil {
+		dep.Spec.Template.Spec.Containers[0].Lifecycle = &corev1.Lifecycle{
+			PostStart: &corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: defaultPostStartCommand,
+				},
+			},
+		}
+		return
+	}
+	if lifecycle.PostStart == nil {
+		lifecycle.PostStart = &corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: defaultPostStartCommand,
+			},
+		}
+		dep.Spec.Template.Spec.Containers[0].Lifecycle = lifecycle
+		return
+	}
+	if lifecycle.PostStart.Exec == nil {
```

Comment:
This is annoying but the Kubernetes API server cannot allow more than one action to be set into a lifecycle. I've tried to do that locally but it returns: `[...] Forbidden: may not specify more than 1 handler type`.

So, unfortunately,  we need to prevent other actions to be set in the post-start hook. In other words, we only support exec action at post-start hook. :disappointed:

Reply:
That's a big problem. I agree with you. I will implement that way 😞

Reply:
I introduced a new type called `NginxLifecycle` with handlers that only accept the Exec action. I don't think we will ever use the official `corev1` Lifecycle unless it supports multiple actions for a given handler.

Reply:
Sure. Great one.